cli/src/pcluster/cli/commands/pcluster3_configuration_converter.py

```diff
@@ -0,0 +1,593 @@
+            return efs_list
+        return None
+
+    def convert_fsx_settings(self, section):
+        fsx_settings = self.config_parser.get(self.cluster_section_name, "fsx_settings", fallback=None)
+        if fsx_settings:
+            for fsx_label in fsx_settings.split(","):
+                fsx_list = []
+                additional_items = [
+                    ("auto_import_policy", "AutoImportPolicy"),
+                    ("fsx_fs_id", "FileSystemId"),
+                    ("fsx_kms_key_id", "KmsKeyId"),
+                    ("storage_capacity", "StorageCapacity", "getint"),
+                    ("deployment_type", "DeploymentType"),
+                    ("storage_type", "StorageType"),
+                    ("imported_file_chunk_size", "ImportedFileChunkSize", "getint"),
+                    ("data_compression_type", "DataCompressionType"),
+                    ("export_path", "ExportPath"),
+                    ("import_path", "ImportPath"),
+                    ("weekly_maintenance_start_time", "WeeklyMaintenanceStartTime"),
+                    (
+                        "automatic_backup_retention_days",
+                        "AutomaticBackupRetentionDays",
+                        "getint",
+                    ),
+                    ("copy_tags_to_backups", "CopyTagsToBackups"),
+                    ("daily_automatic_backup_start_time", "DailyAutomaticBackupStartTime"),
+                    ("per_unit_storage_throughput", "PerUnitStorageThroughput", "getint"),
+                    ("fsx_backup_id", "BackupId"),
+                    ("drive_cache_type", "DriveCacheType"),
+                ]
+                fsx_section, fsx_dict, _section_label = self.convert_storage_base(
+                    "fsx", fsx_label.strip(), additional_items
+                )
+                if fsx_dict:
+                    fsx_section[section] = fsx_dict
+                if fsx_section:
+                    fsx_list.append(fsx_section)
+            return fsx_list
+        return None
+
+    def convert_monitoring(self, section):
+        monitoring = {}
+        logs = {}
+        cloudwatch = {}
+        cw_log_settings = self.config_parser.get(self.cluster_section_name, "cw_log_settings", fallback=None)
+        if cw_log_settings:
+            section_label = f"cw_log {cw_log_settings.strip()}"
+            self.convert_single_field(section_label, "enable", cloudwatch, "Enabled", "getboolean")
+            self.convert_single_field(section_label, "retention_days", cloudwatch, "RetentionInDays", "getint")
+
+        dashboards = {}
+        dasnboard_cloudwatch = {}
+        dashboard_settings = self.config_parser.get(self.cluster_section_name, "dashboard_settings", fallback=None)
+        if dashboard_settings:
+            section_label = f"dashboard {dashboard_settings.strip()}"
+            self.convert_single_field(section_label, "enable", dasnboard_cloudwatch, "Enabled", "getboolean")
+
+        if cloudwatch:
```

Comment:
It seems like this particular logic is repeated many times in the conversion. I would suggest adding a simple helper like:
```
def add_if(self, section, section_name, value):
   if value:
       section[section_name] = value
```

Then the following can become:
```
    self.add_if(monitoring, "Logs", logs)
    self.add_if(dashboards, "CloudWatch", dashboard_cloudwatch)
    self.add_if(monitoring, "Dashboards", dashboards)
    self.add_if(self.pcluster3_configuration, section_name, monitoring)
```

Which seems easier to read because there's less repeated code and less if then, if then, if then...

Reply:
Good idea! Changed